### ISSUE TITLE
Config resets client side after update.

### ISSUE THREAD

**Issue by knoxz** [lorddusk/Bagginses#47]
Why would you disable the loading of the block via config option?
If a server who hosts a pack with your mod now decides to disable bags or just colors.
Clients who do not have the exact same config options can not join the server.
Why not just disable recipes/ nei or jei appearance?

In the latest version of 1.9.4 this a huge issue. The config resets every time I updated the mod clientside. I forgot 1-2 times to change the said config. Uploaded my updated pack to curse via export just to find out, that clients cannot connect to the server because the configs did not match.

A config should be set serverside and then supplied to clients if nesseary. In this instance it should not even be nessesary.

**Comment by knoxz** [comment]
at least this is the first mod I know where you can not connect if the config client and server side does not match.

**Comment by AKTheKnight** [comment]
The way it seems, there are two different issues here:
1) The config resetting when a mod update is applied
and
2) You cannot connect to a server if you have a different config

Now, I agree they are both issues, but they need to be looked at as different things:

The first issue I am not sure of the cause, I will do some checking with the way we manage configs and see if I can spot the issue and fix it somehow.

The second issue is something that cannot be fixed currently. Because of the way most forge mods work, if a block/item is disabled in config we never actually register it. This means that as far as forge and the client know, the item/block doesn't even exist. I cannot then just magically make it appear.

I cannot just "disable recipes" or jei appearance based on config because it's not a simple thing and it's not really worth my time when I have a functioning system now.

Basically I will look into the first issue. The second one I will not be dealing with

**Comment by lorddusk** [comment]
RE 2: That's what I mean, but better worded.

**Comment by knoxz** [comment]
You might want to look how other authors are doing it. Like OpenBlocks for example.
Every block they add can be disabled via config. They use their own Lib with interfaces to make it "easy". [OpenModsLib - Config Section](https://github.com/OpenMods/OpenModsLib/tree/master/src/main/java/openmods/config)
[OpenBlocks Feature Manager](https://github.com/OpenMods/OpenModsLib/blob/f372200faa621c3b832786da01b7e33f6fdc555c/src/main/java/openmods/config/game/ConfigurableFeatureManager.java)
If the block is disabled on the server side and the client can still connect. Even when the he has a different config.
Thats the way every other mod I workd with is doing it.
Well if they give you the option to disable stuff. ;-)

**Comment by madhatr** [comment]
I found this to be very annoying as well.  Took me a while to figure out why I could not connect to my own server.  

 I had been player for a while then noticed oh I don't have any bags.  Check website and see to check config. Check config, see how to enable.  Enable a black bag.  Reset server attempt to connect and Error:"Fatally Missing Blocks and Items".  Decide it's not worth my time atm.   Forward a week or so I try again. I almost removed the mod but I am stubborn and don't like errors.

I think if you have no way to change this or are unwilling to change this a note should be entered in the config file letting server admins know that the client needs to have an identical config.  This will help with new people (like me) figuring out and enjoying your mod.

**Comment by lorddusk** [comment]
In what year do we live? Servers and Clients ALWAYS need to have identical configs. This has been the case since the VERY first FML/Forge server-client connecting... I REALLY don't see the issue here, but that could just be me...
